action/tx_dynamic_fee.go

```diff
@@ -0,0 +1,141 @@
+package action
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/pkg/errors"
+)
+
+// DynamicFeeTx is a transaction type with dynamic fee cap and tip cap
+type DynamicFeeTx struct {
+	chainID    uint32
+	nonce      uint64
+	gasLimit   uint64
+	gasTipCap  *big.Int
+	gasFeeCap  *big.Int
+	accessList types.AccessList
+}
+
+var _ TxCommonWithProto = (*DynamicFeeTx)(nil)
+
+func (tx *DynamicFeeTx) Version() uint32 {
+	return DynamicFeeTxType
+}
+
+func (tx *DynamicFeeTx) ChainID() uint32 {
+	return tx.chainID
+}
+
+func (tx *DynamicFeeTx) Nonce() uint64 {
+	return tx.nonce
+}
+
+func (tx *DynamicFeeTx) Gas() uint64 {
+	return tx.gasLimit
+}
+
+func (tx *DynamicFeeTx) GasTipCap() *big.Int {
+	return tx.gasTipCap
+}
+
+func (tx *DynamicFeeTx) GasFeeCap() *big.Int {
+	return tx.gasFeeCap
+}
+
+func (tx *DynamicFeeTx) AccessList() types.AccessList {
+	return tx.accessList
+}
+
+func (tx *DynamicFeeTx) GasPrice() *big.Int {
+	return tx.gasFeeCap
+}
+
+func (tx *DynamicFeeTx) BlobGas() uint64 { return 0 }
+
+func (tx *DynamicFeeTx) BlobGasFeeCap() *big.Int { return nil }
+
+func (tx *DynamicFeeTx) BlobHashes() []common.Hash { return nil }
+
+func (tx *DynamicFeeTx) BlobTxSidecar() *types.BlobTxSidecar { return nil }
+
+func (tx *DynamicFeeTx) SanityCheck() error {
+	if tx.gasTipCap == nil || tx.gasFeeCap == nil {
+		return ErrMissRequiredField
+	}
+	if tx.gasTipCap.Sign() < 0 {
+		return ErrNegativeValue
+	}
+	if tx.gasFeeCap.Sign() < 0 {
+		return ErrNegativeValue
+	}
+	if tx.gasFeeCap.Cmp(tx.gasTipCap) < 0 {
+		return ErrGasTipOverFeeCap
+	}
+	return nil
+}
+
+func (tx *DynamicFeeTx) toProto() *iotextypes.ActionCore {
+	actCore := iotextypes.ActionCore{
+		Version:  uint32(DynamicFeeTxType),
+		Nonce:    tx.nonce,
+		GasLimit: tx.gasLimit,
+		ChainID:  tx.chainID,
+	}
+	if tx.gasFeeCap != nil {
+		actCore.GasFeeCap = tx.gasFeeCap.String()
+	}
+	if tx.gasTipCap != nil {
+		actCore.GasTipCap = tx.gasTipCap.String()
+	}
+	if len(tx.accessList) > 0 {
+		actCore.AccessList = toAccessListProto(tx.accessList)
+	}
+	return &actCore
+}
+
+func (tx *DynamicFeeTx) fromProto(pb *iotextypes.ActionCore) error {
```

Comment:
let's use `fromProtoDynamicFeeTx(pb *iotextypes.ActionCore) (*DynamicFeeTx, error)` to be consistent

Reply:
I think this is more fit for OO programming ideas. Other 'tx' can be changed to this in the subsequent PRs